src/granite/query/builder.cr

```diff
@@ -16,41 +16,57 @@
 # or
 # - Model.where(field: value).or { whehre(field3: value3) }
 class Granite::Query::Builder(Model)
-  alias FieldName = String
-  alias FieldData = DB::Any
+  enum DbType
+    Mysql
+    Sqlite
+    Pg
+  end
 
   enum Sort
     Ascending
     Descending
   end
 
-  getter where_fields = {} of FieldName => FieldData
+  getter db_type : DbType
+  getter where_fields = [] of NamedTuple(field: String, operator: Symbol, value: DB::Any)
   getter order_fields = [] of NamedTuple(field: String, direction: Sort)
   getter offset : Int64?
   getter limit : Int64?
 
-  def initialize(@boolean_operator = :and)
+  def initialize(@db_type, @boolean_operator = :and)
   end
 
-  def assembler
-    # when adapter.postgresql?
-    Assembler::Postgresql(Model).new self
-    # when adapter.mysql?
-    # etc
+  def assembler : Assembler::Base(Model)
+    case @db_type
+    when DbType::Pg
+      Assembler::Pg(Model).new self
+    when DbType::Mysql
+      Assembler::Mysql(Model).new self
+    when DbType::Sqlite
+      Assembler::Sqlite(Model).new self
+    else
+      raise "Database type not supported"
+    end
```

Comment:
should we memoize this?  Seems like it will create a new one every time its called.